fedimint-api/src/tiered_multi.rs

```diff
@@ -141,19 +142,42 @@ where
 }
 
 impl TieredMulti<()> {
-    // TODO: move somewhere else?
-    pub fn represent_amount<K>(mut amount: Amount, tiers: &Tiered<K>) -> TieredMulti<()> {
-        let coins = tiers
-            .tiers()
-            .rev()
-            .map(|&amount_tier| {
-                let res = amount / amount_tier;
-                amount %= amount_tier;
-                (amount_tier, vec![(); res as usize])
-            })
-            .collect();
+    /// Determines the denominations to use when representing an amount
+    ///
+    /// Algorithm tries to leave the user with a target number of `denomination_sets` starting
+    /// at the lowest denomination.  `self` gives the denominations that the user already has.
+    pub fn represent_amount<K, V>(
+        mut amount: Amount,
+        current_denominations: &TieredMulti<V>,
+        tiers: &Tiered<K>,
+        denomination_sets: u16,
+    ) -> TieredMulti<()> {
+        let mut denominations: Vec<(Amount, ())> = vec![];
+        let mut tiers: Vec<Amount> = tiers.tiers().cloned().collect();
+        tiers.sort();
```

Comment:
This doesn't seem necessary. `Tiered` already uses a `BTreeMap` for this reason (should be documented in `Tiered::tiers` though). It even returns a double ended iterator, so you can call `rev()` on it further down.

Reply:
Okay, I was wondering about whether that was guaranteed.

Reply:
Yep https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.iter